mozilla/bedrock
# [bug 1319244] Test developer and general search

## Description
This adds developer and general custom search to several pages:

Developer search
* /firefox/developer?v=b
* /firefox/channel/desktop?v=b
* /technology

General search
* /contribute
* /404

It also makes some updates to the activist search, moving the code snippet from an include to a macro so it's more reusable and refactors the JS a bit so it's also more reusable.

New switches added:
* `mozorg-search-general`
* `mozorg-search-developer`

Also see https://bugzilla.mozilla.org/show_bug.cgi?id=1323315

## Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1319244

## Testing
* Search should only be visible in English locales.
* Ensure search appears and functions as expected on all target pages (and that I didn't break the earlier activist search on /about etc).
* We're using Traffic Cop for a 50/50 split on both /firefox/developer and /firefox/channel/desktop so it should only be visible on those pages with a `?v=b` param.

## Checklist
